stdlib/public/core/StringComparison.swift

```diff
@@ -0,0 +1,1368 @@
+    start: UnsafeMutableRawPointer(ptr).assumingMemoryBound(to: U.self),
+    count: count
+  )
+}
+@inline(__always)
+private func _unsafeBufferPointerCast<T, U>(
+  _ ptr: UnsafePointer<T>,
+  _ count: Int,
+  to: U.Type = U.self
+) -> UnsafeBufferPointer<U> {
+  return UnsafeBufferPointer(
+    start: UnsafeRawPointer(ptr).assumingMemoryBound(to: U.self),
+    count: count
+  )
+}
+
+// TODO These still crash the compiler
+// TODO(Recursive protocol constraints): Use Base.Element instead of UIntX
+// extension RandomAccessSlice where T == UnsafeBufferPointer<UInt16> {
+//   var rebased: UnsafeBufferPointer<UInt16> {
+//     return UnsafeBufferPointer(rebasing: self)
+//   }
+// }
+// extension RandomAccessSlice where T == UnsafeBufferPointer<UInt8> {
+//   var rebased: UnsafeBufferPointer<UInt8> {
+//     return UnsafeBufferPointer(rebasing: self)
+//   }
+// }
+// extension MutableRandomAccessSlice
+// where T == UnsafeMutableBufferPointer<UInt16> {
+//   var rebased: UnsafeMutableBufferPointer<UInt16> {
+//     return UnsafeMutableBufferPointer(rebasing: self)
+//   }
+// }
+
+// extension Slice where Base == UnsafeBufferPointer<UInt16> {
+//   var rebased: UnsafeBufferPointer<UInt16> {
+//     return UnsafeBufferPointer(rebasing: self)
+//   }
+// }
+//
+// extension Slice where Base == UnsafeMutableBufferPointer<UInt16> {
+//   var rebased: UnsafeMutableBufferPointer<UInt16> {
+//     return UnsafeMutableBufferPointer(rebasing: self)
+//   }
+// }
+
+// extension Slice where Base == UnsafeMutableBufferPointer<Base.Element> {
+//   var rebased: UnsafeMutableBUfferPointer<Base.Element> {
+//     return UnsafeMutableBufferPointer(rebasing: self)
+//   }
+// }
+
+internal let _leadingSurrogateBias: UInt16 = 0xd800
+internal let _trailingSurrogateBias: UInt16 = 0xdc00
+internal let _surrogateMask: UInt16 = 0xfc00
+
+@inline(__always)
+internal func _isSurrogate(_ cu: UInt16) -> Bool {
+  return _isLeadingSurrogate(cu) || _isTrailingSurrogate(cu)
+}
+
+@inline(__always)
+internal func _isLeadingSurrogate(_ cu: UInt16) -> Bool {
+  // NOTE: Specifically match against the trailing surrogate mask, as it matches
+  // more cases.
+  return cu & _surrogateMask == _leadingSurrogateBias
+}
+
+@inline(__always)
+internal func _isTrailingSurrogate(_ cu: UInt16) -> Bool {
+  return cu & _surrogateMask == _trailingSurrogateBias
+}
+@inline(__always)
+internal func _decodeSurrogatePair(
+  leading high: UInt16, trailing low: UInt16
+) -> UInt32 {
+  _sanityCheck(_isLeadingSurrogate(high) && _isTrailingSurrogate(low))
+  let hi10: UInt32 = UInt32(high) &- UInt32(_leadingSurrogateBias)
+  _sanityCheck(hi10 < 1<<10, "I said high 10. Not high, like, 20 or something")
+  let lo10: UInt32 = UInt32(low) &- UInt32(_trailingSurrogateBias)
+  _sanityCheck(lo10 < 1<<10, "I said low 10. Not low, like, 20 or something")
+
+  let result = ((hi10 &<< 10) | lo10) &+ 0x1_00_00
+  // print("""
+  //   hi10: \(hex(hi10)), lo10: \(hex(lo10)) ==> \(hex(result))
+  // """)
```

Comment:
Please remove commented out code. If you want to keep this kind of thing around for future debugging purposes, then we should come up with a different scheme for that.